static/app/components/globalDrawer/components.tsx

```diff
@@ -28,35 +36,193 @@ interface DrawerPanelProps {
   children: React.ReactNode;
   headerContent: React.ReactNode;
   onClose: DrawerContentContextType['onClose'];
+  drawerKey?: string;
   drawerWidth?: DrawerOptions['drawerWidth'];
   transitionProps?: AnimationProps['transition'];
 }
 
+function getDrawerWidthKey(drawerKey: string) {
+  return `drawer-width:${drawerKey}`;
+}
+
 export function DrawerPanel({
   ref,
   ariaLabel,
   children,
   transitionProps,
   onClose,
   drawerWidth,
+  drawerKey,
 }: DrawerPanelProps & {
   ref?: React.Ref<HTMLDivElement>;
 }) {
+  const resizeHandleRef = useRef<HTMLDivElement>(null);
+  const panelRef = useRef<HTMLDivElement>(null);
+  const rafIdRef = useRef<number | null>(null);
+  const initialMousePositionRef = useRef<number | null>(null);
+  const isResizingRef = useRef<boolean>(false);
+
+  // Calculate initial width from props or default to 50%
+  const calculateInitialWidth = (savedWidth?: number) => {
+    // If we have a saved width, use it but ensure it's within bounds
+    if (savedWidth !== undefined) {
+      const minWidthPercent = 30;
+      const maxWidthPercent = 90;
+      return Math.min(Math.max(savedWidth, minWidthPercent), maxWidthPercent);
+    }
+
+    if (drawerWidth) {
+      // If width is already in percentage, parse and clamp it
+      if (drawerWidth.endsWith('%')) {
+        const parsedPercent = parseFloat(drawerWidth);
+        const minWidthPercent = 30;
+        const maxWidthPercent = 90;
+        return Math.min(Math.max(parsedPercent, minWidthPercent), maxWidthPercent);
+      }
+      // If width is in pixels, convert to percentage and clamp
+
+      const viewportWidth = typeof window === 'undefined' ? 1000 : window.innerWidth;
+      const parsedPixels = parseFloat(drawerWidth);
+      const percentValue = (parsedPixels / viewportWidth) * 100;
+      const minWidthPercent = 30;
+      const maxWidthPercent = 90;
+      return Math.min(Math.max(percentValue, minWidthPercent), maxWidthPercent);
+    }
+
+    return 50; // Default to 50%
+  };
+
+  // Store persisted width in localStorage, but don't use it for rendering state
+  const [persistedWidthPercent, setPersistedWidthPercent] =
+    useSyncedLocalStorageState<number>(
+      drawerKey ? getDrawerWidthKey(drawerKey) : 'drawer-width:default',
+      (value?: unknown) => {
+        const savedWidth = typeof value === 'number' ? value : undefined;
+        return calculateInitialWidth(savedWidth);
+      }
```

Comment:
Why does the type have to be `unknown` here?

Reply:
The type signature of that hook is
```
export function useSyncedLocalStorageState<S>(
  key: string,
  initialState: S | ((value?: unknown, rawValue?: unknown) => S)
): [S, (value: S) => void] {
```